interfaces/apparmor/template.go

```diff
@@ -550,11 +437,293 @@ var defaultTemplate = `
+  /{,usr/}bin/touch ixr,
+  /{,usr/}bin/tput ixr,
+  /{,usr/}bin/tr ixr,
+  /{,usr/}bin/true ixr,
+  /{,usr/}bin/tty ixr,
+  /{,usr/}bin/uname ixr,
+  /{,usr/}bin/uniq ixr,
+  /{,usr/}bin/unlink ixr,
+  /{,usr/}bin/unxz ixr,
+  /{,usr/}bin/unzip ixr,
+  /{,usr/}bin/uptime ixr,
+  /{,usr/}bin/vdir ixr,
+  /{,usr/}bin/wc ixr,
+  /{,usr/}bin/which ixr,
+  /{,usr/}bin/xargs ixr,
+  /{,usr/}bin/xz ixr,
+  /{,usr/}bin/yes ixr,
+  /{,usr/}bin/zcat ixr,
+  /{,usr/}bin/z{,e,f}grep ixr,
+  /{,usr/}bin/zip ixr,
+  /{,usr/}bin/zipgrep ixr,
+
+  # lsb-release
+  /usr/bin/lsb_release ixr,
+  /usr/bin/ r,
+  /usr/share/distro-info/*.csv r,
+
+  # For printing the cache (we don't allow updating the cache)
+  /{,usr/}sbin/ldconfig{,.real} ixr,
+
+  # Allow all snaps to chroot
+  /{,usr/}sbin/chroot ixr,
+`
+
+// defaultTemplate contains the default apparmor template for core* bases. It
+// can be overridden for testing using MockTemplate().
+var defaultTemplate = templateCommon + defaultTemplateRules + templateFooter
+
+// defaultBaseTemplateRules for non-core* bases. When a snap specifies an
+// alternative base to core*, it is allowed read-only access to all files
+// within the base, but all other accesses (eg, host file rules, signal, dbus,
+// unix, etc rules) should be the same as the default template.
+//
+// For clarity and ease of maintenance, we will whitelist top-level directories
+// here instead of using glob rules (we can add more if specific bases
+// dictate).
+var defaultBaseTemplateRules = `
+  # Default rules for non-core base runtimes
+
+  # /bin and /sbin (/usr/{,local/}{s,bin} handled in /usr)
+  /{,s}bin/ r,
+  /{,s}bin/** mrklix,
+
+  # /lib - the mount setup may bind mount to:
+  #
+  # - /lib/firmware
+  # - /lib/modules
+  #
+  # Everything but /lib/firmware and /lib/modules
+  /lib/ r,
+  /lib/[^fm]** mrklix,
+  /lib/{f[^i],m[^o]}** mrklix,
+  /lib/{fi[^r],mo[^d]}** mrklix,
+  /lib/{fir[^m],mod[^u]}** mrklix,
+  /lib/{firm[^w],modu[^l]}** mrklix,
+  /lib/{firmw[^a],modul[^e]}** mrklix,
+  /lib/{firmwa[^r],module[^s]}** mrklix,
+  /lib/modules[^/]** mrklix,
+  /lib/firmwar[^e]** mrklix,
+  /lib/firmware[^/]** mrklix,
+
+  # /lib64, etc
+  /lib[^/]** mrklix,
+
+  # /opt
+  /opt/ r,
+  /opt/** mrklix,
+
+  # /usr - the mount setup may bind mount to:
+  #
+  # - /usr/lib/modules
+  # - /usr/lib/firmware
+  # - /usr/lib/snapd
+  # - /usr/src
+  #
+  # Everything but /usr/lib and /usr/src, which are handled elsewhere.
+  /usr/ r,
+  /usr/[^ls]** mrklix,
+  /usr/{l[^i],s[^r]}** mrklix,
+  /usr/{li[^b],sr[^c]}** mrklix,
+  /usr/{lib,src}[^/]** mrklix,
+  # Everything in /usr/lib except /usr/lib/firmware, /usr/lib/modules and
+  # /usr/lib/snapd, which are handled elsewhere.
+  /usr/lib/ r,
+  /usr/lib/[^fms]** mrklix,
+  /usr/lib/{f[^i],m[^o],s[^n]}** mrklix,
+  /usr/lib/{fi[^r],mo[^d],sn[^a]}** mrklix,
+  /usr/lib/{fir[^m],mod[^u],sna[^p]}** mrklix,
+  /usr/lib/{firm[^w],modu[^l],snap[^d]}** mrklix,
+  /usr/lib/snapd[^/]** mrklix,
```

Comment:
Well, yeah, it is awkward cause snapd, firmware and modules are all different lengths. The idea was that snapd is dropping out of the alternation at this point, so I put it there. If I put it at the end, it might look weird that it dropped out of the alternation.